### ISSUE TITLE
npm install time

### ISSUE THREAD

**Issue by acoffin** [filamentgroup/grunticon#268]
I'm new to grunt but thus far have only had trouble installing grunticon using "npm install grunt-grunticon". I've run it with -dd and it seems to create directories and sub directories ad infinitum to the point where I can no longer even delete the grunt-grunticon folder under node-modules because the file path is too long for Windows to handle.

Is grunticon supposed to take a really long time to install and these long file paths are intended, or is there something incorrect with my setup?

EDIT:
It seems like I could get it to complete the install, it just took 10-20 minutes. It may be that there is an issue installing phantomjs rather than grunticon. I don't know enough to really say much for certain.

**Comment by jefflembeck** [comment]
@acoffin i must say that this is bizarre and something I've never seen before. If you have any environment things to add, it might be useful in case it comes up again.

**Comment by acoffin** [comment]
I just closed this because I started to think it was normal. All the other packages I've installed took seconds, grunticon is taking upwards of 10 minutes. I take it that's not normal?

I am definitely having a problem getting phantomjs to install. I'm running through the whole grunticon install process again to have a clean error log to post.

**Comment by acoffin** [comment]
The debug.log is too long to post as a comment here. Here is where the error occurs:

```
230492 verbose pkgid phantomjs@1.9.17
230493 verbose cwd E:\grunticon-test
230494 error Windows_NT 6.1.7601
230495 error argv "C:\\Program Files\\nodejs\\\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install"
230496 error node v0.12.4
230497 error npm  v2.10.1
230498 error code ELIFECYCLE
230499 error phantomjs@1.9.17 install: `node install.js`
230499 error Exit status 1
230500 error Failed at the phantomjs@1.9.17 install script 'node install.js'.
230500 error This is most likely a problem with the phantomjs package,
230500 error not with npm itself.
230500 error Tell the author that this fails on your system:
230500 error     node install.js
230500 error You can get their info via:
230500 error     npm owner ls phantomjs
230500 error There is likely additional logging output above.
230501 verbose exit [ 1, true ]
```

This happens right after "optipng pre-build test passed successfully"

**Comment by jefflembeck** [comment]
@acoffin are you able to just install the `phantomjs` package?

`npm install phantomjs`

**Comment by acoffin** [comment]
I tried installing to my local drive this morning and didn't have any problems with phantomjs or grunticon. I did get the same errors when installing to a local server mapped as a network drive. Could this be the issue?

I'm currently copying the properly installed grunticon and phantomjs files to my server location and running npm install to see if it likes what it sees.

**Comment by jefflembeck** [comment]
@acoffin that _might_ be the issue. The phantomjs installer does some magic with installations and paths that might be affected

**Comment by acoffin** [comment]
It looks like installing grunticon on the local drive and copying it to my project node_modules folder did the trick. Running "npm install" saw it as already installed and gave no errors.

The funny thing is, I tried install phantomjs and grunticon last week on the local drive and got the same errors I was getting in the network location. I only tried again this morning so I'd have shorter file paths in the debug.log I was going to submit. Glad I didn't spend another day troubleshooting the network location.

Thanks for your help!
